src/backend/joanie/core/api/client/__init__.py

```diff
@@ -541,6 +542,58 @@ def withdraw(self, request, pk=None):  # pylint: disable=no-self-use, invalid-na
 
         return Response(status=HTTPStatus.NO_CONTENT)
 
+    # @extend_schema(
+    #     request=None,
+    #     responses={
+    #         204: OpenApiTypes.NONE,
+    #         422: serializers.ErrorResponseSerializer,
+    #     },
+    # )
+    @action(detail=True, methods=["POST"])
+    def submit_installment_payment(self, request, pk=None):  # pylint: disable=unused-argument
+        """
+        Pay a failed installment that was scheduled for a given order.
+        """
+        order = self.get_object()
+        if order.state != enums.ORDER_STATE_FAILED_PAYMENT:
+            return Response(
+                {
+                    "detail": "The order is not in failed payment state for last payment."
+                },
+                status=HTTPStatus.UNPROCESSABLE_ENTITY,
+            )
+
+        payment_backend = get_payment_backend()
+        installment = order.get_first_installment_refused()
+
+        card_id = request.data.get("card_id")
+        if not card_id:
+            payment_infos = payment_backend.create_payment(
+                order=order,
+                billing_address=order.main_invoice.recipient_address,
+                installment=installment,
+            )
+            return Response(payment_infos, status=HTTPStatus.OK)
+
+        try:
+            credit_card = CreditCard.objects.get_card_for_owner(
+                pk=card_id, user_id=order.owner.id
+            )
+        except CreditCard.DoesNotExist:
+            return Response(
+                {"detail": "Credit card does not exist."},
+                status=HTTPStatus.NOT_FOUND,
+            )
```

Comment:
I wonder if we should not return a regular payment in this case? WDYT?

Reply:
It's strange to do that.

Reply:
I've changed the queryset in the CreditCardManager to not raise CreditCardDoesNotExist, and instead use `.filter` and `.first()`